plugins-builtin/src/main/scala/za/co/absa/enceladus/plugins/builtin/common/mq/kafka/KafkaSecurityParams.scala

```diff
@@ -27,8 +27,12 @@ case class KafkaSecurityParams(
                                 saslMechanism: Option[String]
                               ) {
   def toMap: Map[String, String] = {
-    Map(SecurityProtocolKey -> securityProtocol) ++
-      saslMechanism.fold(Map.empty[String, String])(saslValue => Map(SaslMechanismKey -> saslValue))
+    Map(
+      SecurityProtocolKey -> Some(securityProtocol),
+      SaslMechanismKey -> saslMechanism
+    ).collect {
+      case (key, Some(value)) => key -> value
+    }
```

Comment:
Did this in the first iteration when I added 3 more params. I would just leave it be if no one has any comments.

Reply:
There is also this option to use:
`for ((k, Some(v)) <- map) yield k -> v`
But I would not say it's that much better

Reply:
I think it's OK , how it is.